### PR TITLE
Sorting categories body in the observable

### PR DESCRIPTION
**Pull request by shazm**
#### Summary

Updates channel-list sorting inside categories:

- Alpha sort now handles the channel being muted
- Sorting has been moved from component to the withObservable HOC

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-42165

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

File: app/components/channel_list/categories/categories.tsx [hunk 1]
```diff
@@ -20,23 +21,27 @@ const styles = StyleSheet.create({
...
-        </>
-    );
-};
-
 const Categories = (props: Props) => {
-    if (!props.categories.length) {
-        return <LoadCategoriesError/>;
-    }
+    const intl = useIntl();
+    const renderCategory = useCallback((data: {item: CategoryModel}) => {
```
Comment (avinashlng1080) [inline app/components/channel_list/categories/categories.tsx:26]: You can move this `renderCategory` method outside of the `Categories` component.  

The deps array is watching for `props.categories` whereas it should be watching for the `data.item` ...but moving it outside would solve it.

File: app/components/channel_list/categories/categories.tsx [hunk 2]
```diff
@@ -20,23 +21,27 @@ const styles = StyleSheet.create({
...
-const renderCategory = (data: {item: CategoryModel}) => {
-    return (
-        <>
-            <CategoryHeader category={data.item}/>
-            <CategoryBody category={data.item}/>
-        </>
-    );
-};
-
 const Categories = (props: Props) => {
```
Comment (avinashlng1080) [inline app/components/channel_list/categories/categories.tsx:24]: It's not a standard/convention but all our components destructure its props at this line and thus we avoid `props.categories`.  0/5 on this one.

File: app/components/channel_list/categories/body/index.ts [hunk 3]
```diff
@@ -1,17 +1,95 @@
...
+    switch (category.sorting) {
+        case 'alpha': {
+            const channels = category.channels.observeWithColumns(['display_name']);
+            const settings = channels.pipe(
+                switchMap((cs) => querySettings(database, cs)),
+            );
+            return combineLatest([channels, settings]).pipe(
+                switchMap(([cs, st]) => buildAlphaData(cs, st, locale)),
+            );
+            break;
```
Comment (avinashlng1080) [inline app/components/channel_list/categories/body/index.ts:69]: can be removed

**Review by avinashlng1080 — CHANGES_REQUESTED** [review]
PR looks good.  Just one nit.

**Review by larkox — APPROVED** [review]
Apart of the other comments, LGTM.

**Comment by enahum** [comment]
merging

File: app/components/channel_list/categories/categories.tsx (hunk 1 above)
Comment (avinashlng1080) [reply app/components/channel_list/categories/categories.tsx:29]: Now, that's it using useCallback, it's fine.

File: app/components/channel_list/categories/categories.tsx [hunk 4]
```diff
@@ -22,26 +24,31 @@ const styles = StyleSheet.create({
...
+                    locale={intl.locale}
+                />
+            </>
+        );
+    }, [categories, currentChannelId, intl.locale]);
 
-const Categories = (props: Props) => {
-    if (!props.categories.length) {
+    // Sort Categories
+    categories.sort((a, b) => a.sortOrder - b.sortOrder);
```
Comment (avinashlng1080) [inline app/components/channel_list/categories/categories.tsx:43]: I know the PR has already been merged.  But the `categories`, could they be sorted within the `withObservables` block under   `app/components/channel_list/categories/index.ts`  ? That would avoid sorting it each time the component renders?  @shazm

Comment (enahum) [reply app/components/channel_list/categories/categories.tsx:43]: The thing is that you want them to be sorted each time the component re-renders

**Pull request merged**